File: client/src/main/java/au/com/codeka/warworlds/client/game/starsearch/StarSearchScreen.java
package au.com.codeka.warworlds.client.game.starsearch;

import android.view.ViewGroup;

import au.com.codeka.warworlds.client.game.solarsystem.SolarSystemScreen;
import au.com.codeka.warworlds.client.ui.Screen;
import au.com.codeka.warworlds.client.ui.ScreenContext;
import au.com.codeka.warworlds.client.ui.ShowInfo;

public class StarSearchScreen extends Screen {
  private StarSearchLayout layout;

  @Override
  public void onCreate(ScreenContext context, ViewGroup container) {
    super.onCreate(context, container);
    layout =
        new StarSearchLayout(
            context.getActivity(),
            star -> context.pushScreen(new SolarSystemScreen(star, -1)));
  }

  @Override
  public ShowInfo onShow() {
    return ShowInfo.builder().view(layout).build();
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/game/starsearch/StarSearchScreen.kt
package au.com.codeka.warworlds.client.game.starsearch

import android.view.ViewGroup
import au.com.codeka.warworlds.client.game.solarsystem.SolarSystemScreen
import au.com.codeka.warworlds.client.ui.Screen
import au.com.codeka.warworlds.client.ui.ScreenContext
import au.com.codeka.warworlds.client.ui.ShowInfo
import au.com.codeka.warworlds.client.ui.ShowInfo.Companion.builder
import au.com.codeka.warworlds.common.proto.Star

class StarSearchScreen : Screen() {
  private lateinit var layout: StarSearchLayout

  override fun onCreate(context: ScreenContext, container: ViewGroup) {
    super.onCreate(context, container)
    layout = StarSearchLayout(
        context.activity,
        object : StarSearchLayout.Callback {
          override fun onStarClick(star: Star?) {
            context.pushScreen(SolarSystemScreen(star!!, -1))
          }
        })
  }

  override fun onShow(): ShowInfo? {
    return builder().view(layout).build()
  }
}